values.yaml

```diff
@@ -328,10 +328,24 @@ syncCatalog:
   # to all namespaces.
   k8sSourceNamespace: null
 
-  # consulPrefix is the service prefix which preprends itself
+  # addK8SNamespaceSuffix appends Kubernetes namespace suffix to
+  # each service name synced to Consul, separated by an underscore.
+  # For example, for a service 'foo' in the default namespace,
+  # the sync process will create a Consul service named 'foo_default'.
+  # Set this flag to true to avoid registering services with the same name
+  # but in different namespaces as instances for the same Consul service.
+  # Namespace suffix is not added if 'annotationServiceName' is provided.
+  # NOTE: Setting this property to 'true' will result in deregistering of existing
+  # synced services in Consul and registering them with a new name.
+  addK8SNamespaceSuffix: false
```

Comment:
I think we should set this to `true` by default. The current default has the major problem that a service with the same name in any namespace will be synced to the same Consul service.

We'd need to document the breaking change and tell people how to upgrade safely (set this value to false) but I think it's worth it to not put new users in a bad state.

Reply:
@lkysow After thinking about this more, I agree that this should be true by default and that it's worth making this a breaking change.